### ISSUE TITLE
Using autoclassed concrete classes, and pyjnius is complaining that my interface?? objects have no attribute (which are methods on the concrete class)

### ISSUE THREAD

**Issue by coderextreme** [kivy/pyjnius#270]
Sample project is here https://github.com/coderextreme/pythonSAI.  I think I may need to use more than autoclass

**Comment by coderextreme** [comment]
Traceback (most recent call last):
  File "abox.py", line 60, in <module>
    ProtoInstance29.setIS(IS30)
AttributeError: 'org.web3d.x3d.sai.Core.ProtoInstance' object has no attribute 'setIS'

and similar

**Comment by JPDSousa** [comment]
Any update on this front? I'm having the same issue 😢

**Comment by coderextreme** [comment]
I’m guessing it’s related to the getSuperclass call in pyjnius (in the autoclass function I think).

Make sure you are enforcing your interface contracts.  Do not return a concrete class, but rather return the interface (or make the types returned match).

John

Sent from Mail for Windows 10

**Comment by JPDSousa** [comment]
I caught this issue while doing a `getClass` over an interface object

**Comment by coderextreme** [comment]
I don’t think there’s such a thing as an interface object???  I know what
you mean though.

**Comment by JPDSousa** [comment]
What I mean is that I have a given class that returns an interface type and for some reason I cannot do a getClass over that method, as it throws a no attribute error. The strange thing is that find_javaclass.isInstance does not work as well, as it always returns false, even when I load the interface type.

**Comment by coderextreme** [comment]
Yeah, you’ll have to dig into pyjnius I guess, if you don’t want to change your code. You can call a similar method with a different signature in your implementing class, which returns an instance of a concrete class which inherits from the interface, but does not declare the similar method in the interface.   GL  What I am doing is implementing primitive type wrappers around my parameters, so that the concrete class method is selected instead of the interface method.  It’s like you’re ignoring the interface though.

I do not like it. If your implementing class does return an interface type just like the interface does, I suggest that the super class is overriding the subclass and blocking the implementation, but I haven’t proved anything satisfactorily yet.  I need to start up a debugger, but not sure which one to use with kivy.

Basically, my “boss” is overly optimistic, and I am pessimistic because I am just a newb at Python.   We need help from the pyjnius experts!

Another solution is not to use return values.  YMMV

John

Sent from Mail for Windows 10

**Comment by carlsonsolutiondesign** [comment]
Also complains on abstract classes in addition to interfaces
